### DISCUSSION TITLE
jq expression may result in non-json or multiple items

### DISCUSSION THREAD

**Discussion by globalflea** [serverlessworkflow/specification#553] [Q&A]
AFAIK, jq filters may give results thats not legal json or multiple distinct json results.  For non-json results, is this allowed? And what do you do for multiple json results.  

Examples from online jq manual:
non-json output:
```
jq '.[]'
Input	{"a": 1, "b": 1}
Output	
1
1
``` 

multiple json outputs:
```
jq '{user, title: .titles[]}'
input.     {"user":"stedolan","titles":["JQ Primer", "More JQ"]}
output
{"user":"stedolan", "title": "JQ Primer"}
{"user":"stedolan", "title": "More JQ"}

```

**Comment by ricardozanini** [comment]
In this case, the runtime should apply the fix to the result.

In the first scenario, you would have an array, like: `[1, 1]`. In the latter, again, an array of objects. The most specific outcome is to merge the results in a JSON array. But I agree that the spec should add a note on how to handle such merging cases. @tsurdilo wdyt?
